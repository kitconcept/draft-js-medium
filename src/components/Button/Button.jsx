/**
 * Button component.
 * @module components/Button
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

/**
 * Button component class.
 * @class Button
 * @extends Component
 */
export default class Button extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    icon: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
  };

  /**
   * Render method.
   * @function render
   * @returns {string} Markup of the container.
   */
  render() {
    return (
      <button className="ui icon button" onClick={this.props.onClick}>
        <i aria-hidden="true" className={`${this.props.icon} icon`} />
      </button>
    );
  }
}
