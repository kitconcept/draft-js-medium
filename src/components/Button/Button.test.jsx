import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Button', () => {
  it('renders a button component', () => {
    const component = renderer.create(
      <Button icon="bold" onClick={() => {}} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
