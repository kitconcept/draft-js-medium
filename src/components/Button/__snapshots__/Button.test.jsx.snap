// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button component 1`] = `
<button
  className="ui icon button"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="bold icon"
  />
</button>
`;
